### PR TITLE
Improve PHP 8.3 Compatibility

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

See also

* https://github.com/ampproject/amp-toolbox-php/pull/545

## Checklist

- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by swissspidy** [comment]
All PHP warnings seem to have been gone.

The failing tests are unrelated I think, as WP 6.4 happened in the meantime.

**Comment by westonruter** [comment]
Looks like `Test_AMP_Validation_Manager::test_add_block_source_comments` is failing due to an update in Gutenberg.

File: tests/php/src/Helpers/PrivateAccess.php [hunk 1]
```diff
@@ -59,12 +59,11 @@ private function call_private_static_method( $class, $method_name, $args = [] )
 	 * @throws ReflectionException If the object could not be reflected upon.
 	 */
 	private function set_private_property( $object, $property_name, $value ) {
-		$property = ( new ReflectionClass( $object ) )->getProperty( $property_name );
+		$reflection_class = new ReflectionClass( $object );
+		$property = $reflection_class->getProperty( $property_name );
```
Comment (westonruter) [inline tests/php/src/Helpers/PrivateAccess.php]: Nit for PHPCS:

```suggestion
		$reflection_class = new ReflectionClass( $object );
		$property         = $reflection_class->getProperty( $property_name );
```

File: tests/php/src/Helpers/PrivateAccess.php [hunk 2]
```diff
@@ -59,12 +59,11 @@ private function call_private_static_method( $class, $method_name, $args = [] )
...
 		$property->setAccessible( true );
 
-		// Note: In PHP 8, `ReflectionProperty::getValue()` now requires that an object be supplied if it's a
-		// non-static property.
-		$property->isStatic() ? $property->setValue( $value ) : $property->setValue( $object, $value );
+		$property->isStatic() ? $reflection_class->setStaticPropertyValue( $property_name, $value ) : $property->setValue( $object, $value );
```
Comment (westonruter) [inline tests/php/src/Helpers/PrivateAccess.php]: Nit, let's take this opportunity to not use a ternary as a bare statement:

```suggestion
		if ( $property->isStatic() ) {
			$reflection_class->setStaticPropertyValue( $property_name, $value );
		} else {
			$property->setValue( $object, $value );
		}
```

**Comment by thelovekesh** [comment]
Tests were failing due to changes introduced by https://github.com/WordPress/wordpress-develop/commit/ac8bca4fdf in core and by https://github.com/WordPress/gutenberg/pull/55240 in GB plugin 17.1.0+.

**Review by westonruter — APPROVED** [review]

**Review by westonruter — APPROVED** [review]
Thank you both!

**Pull request merged**